### PR TITLE
credentials for services migrated to iam

### PR DESCRIPTION
**Pull request by dpopp07**
@germanattanasio I spoke to Taj the other day about this issue. I just took a quick pass at it while I had a free minute. We may not be able to rely on the `resource_name` in `credentials` so this may not be ideal, but it seemed to be that was the best way to support the migration. Let me know what you think.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -27,6 +27,18 @@ module.exports.getCredentials = function(name, plan, iname) {
         }
       }
     }
+    // Check if credentials weren't found because the service was migrated to IAM
+    for (var service_name in services) {
+      for (var i = services[service_name].length - 1; i >= 0; i--) {
+        var instance = services[service_name][i];
+        var usingResourceName = instance.credentials && instance.credentials.resource_name;
+        if ((usingResourceName && instance.credentials.resource_name === name) && 
+          (!iname || iname === instance.name)
```
Comment (germanattanasio) [inline index.js]: Why don't you check 
```js
if (instance.credentials.resource_name === name && (!iname || iname === instance.name)) {
  // ...
}
```

Comment (dpopp07) [reply index.js]: I was just trying to be safe. In the case that `credentials` isn't a key on the object, that would cause the module to crash. But I can make that change

**Comment by germanattanasio** [comment]
@dpopp07 This looks good. 
Here are some tasks to address before merging this PR:

- [ ] Migrate a Discovery or Conversation CF instance and check if the JSON matches the one you are expecting.
- [ ] Add a section in the README about migrated CF services.
- [ ] Add more tests for corner cases where some fields may not exist. Ideally, the new code coverage should be higher than the previous one.

**Comment by dpopp07** [comment]
Closing this PR, as `ibmcloud-link` seems to be going away. I can re-open if the need arises

**Comment by germanattanasio** [comment]
Thanks for the work @dpopp07
